### PR TITLE
revert to using infra id

### PR DESCRIPTION
**Pull request by jim-minter**
Encountered a nasty networking issue.  We are going to need all VM hostnames to be unique, because Azure-provided name resolution does not take subnets into account.

Create a cluster A, then create a cluster B on the same vnet (in different subnets, of course).  Reboot a master node in A.  The node fails to come back up because its sdn fails to come up, because it resolves its IP via DNS to the equivalent master node in the newer cluster B.  The error you get is `HostIP %q for local subnet does not match with nodeIP %q, Waiting for master to update subnet for node %q ...`.

This PR naively reverts our behaviour to match the openshift installer, which inserts a random "infraID" in all Azure resources it creates.  I wonder whether it is better to stay true to the installer, or reduce the scope of this patch to only do this to machine hostnames.

### CODE REVIEW

File: pkg/backend/openshiftcluster/create.go [hunk 1]
```diff
@@ -325,8 +325,8 @@ func getRHCOSImage(ctx context.Context) (*azuretypes.Image, error) {
 	}, nil
 }
 
-func randomLowerCaseAlphanumericString(n int) (string, error) {
-	return randomString("abcdefghijklmnopqrstuvwxyz0123456789", n)
+func randomLowerCaseAlphanumericStringWithNoVowels(n int) (string, error) {
```
Comment (nilsanderselde) [inline pkg/backend/openshiftcluster/create.go:328]: > WithNoVowels

🧐

Comment (jim-minter) [reply pkg/backend/openshiftcluster/create.go:328]: you can't have a naughty word without vowels...

**Comment by mgahagan73** [comment]
I've confirmed that clusters created with this PR provision nodes with the randomized names e.g.:

{
      "count": 1,
      "diskSizeGb": 128,
      "name": "mgahagan-18a-d4ld9-worker-eastus1",
      "subnetId": "/subscriptions/XXXXXXXXX/resourceGroups/v4-eastus/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/mgahagan-18a-worker",
      "vmSize": "Standard_D2s_v3"
    }

After cordoning and draining one master node, I powered off a master node, confirmed it reported NotReady,SchedulingDisabled, powered the node back on and confirmed it reported Ready,SchedulingDisabled, uncordon'ed the node and it went back into Ready status and was accessable. I then repeated the same operation on the second cluster that was created with the same results.

**Pull request merged**